### PR TITLE
Rewrite endecode to remove internal dependences on OSSL

### PR DESCRIPTION
**Pull request by alexzas**
Rewrite of encoders / decoders test to hopefully make it much more simpler and removing dependency on internal OSSL structures.

### CODE REVIEW

File: test/oqs_test_endecode.c [hunk 1]
```diff
@@ -1,68 +1,101 @@
...
+        {"DER", "PrivateKeyInfo",          NULL, NULL,                                   OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+        {"DER", "EncryptedPrivateKeyInfo", NULL, "Pass the holy handgrenade of antioch", OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS},
+        {"DER", "SubjectPublicKeyInfo",    NULL, NULL,                                   OSSL_KEYMGMT_SELECT_PUBLIC_KEY |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+};
+
+static const char *sigalg_names[] = {
+///// OQS_TEMPLATE_FRAGMENT_SIGNATURE_CASES_START
```
Comment (baentsch) [inline test/oqs_test_endecode.c:53]: Is omission of a jinja2 template script intentional? The code in these template brackets would (need to be) generated by a (suitably named "signature_cases.fragment") script in "oqs-template/test/oqs_test_endecode.c". In turn all other fragments in that directory are apparently no longer needed after the re-write. If you'd be willing to dive into this, a correct code change will have been done if "python3 oqs-template/generate.py" created correct code (passing testing, of course) that then also would become part of the PR. If you don't want to get into `jinja2`, no problem, I'll handle then in a followup PR.

**Review by baentsch — APPROVED** [review]
Thanks very much, @alexzas for this re-write: Much cleaner and easier to understand! The dependency is gone for good now as far as I can see. Please let me know if you'd like to see this merged as-is or whether you want to give the `jinja2` templating issue a go (edit/add: I'd otherwise add this in a separate PR).

**Comment by baentsch** [comment]
PS: [This](https://github.com/open-quantum-safe/oqs-provider/blob/fbd2538173bf8f24274c89a94a6f950b65eb928e/test/CMakeLists.txt#L34-L36) `if` statement ideally should fall away either in this PR or #150 (by @distorted-mdw) to fully break the dependency on OpenSSL source code: https://github.com/open-quantum-safe/oqs-provider/blob/fbd2538173bf8f24274c89a94a6f950b65eb928e/test/CMakeLists.txt#L34-L36

**Comment by alexzas** [comment]
@baentsch , thanks! 
Re ${OPENSSL_BLDTOP}, I see the last place OSSL source test files are used are in oqs_groups. 
Specifically, test/certs and test/recipes/90-test_sslapi_data/passwd.txt are passed as inputs to oqs_groups. 
Maybe its worth to copy certs directory and the passwd.txt to oqs-providers/tests to remove this dependency.

Lets do the jinja2 and the above  as part of  separate PRs?

**Pull request merged**